core/src/main/java/com/scalar/db/storage/jdbc/JdbcAdmin.java

```diff
@@ -429,6 +441,62 @@ public TableMetadata getTableMetadata(String namespace, String table) throws Exe
     return builder.build();
   }
 
+  @Override
+  public TableMetadata getImportTableMetadata(String namespace, String table)
+      throws ExecutionException {
+    TableMetadata.Builder builder = TableMetadata.newBuilder();
+    boolean primaryKeyExists = false;
+
+    if (!rdbEngine.isImportable()) {
+      throw new UnsupportedOperationException(
+          "Importing table is not allowed in this storage: " + rdbEngine);
+    }
+
+    try (Connection connection = dataSource.getConnection()) {
+      if (!tableExistsInternal(connection, namespace, table)) {
+        throw new IllegalArgumentException(
+            "The table " + getFullTableName(namespace, table) + "  does not exist");
+      }
+
+      DatabaseMetaData metadata = connection.getMetaData();
+      ResultSet resultSet = metadata.getPrimaryKeys(null, namespace, table);
+      while (resultSet.next()) {
+        primaryKeyExists = true;
+        String columnName = resultSet.getString(JDBC_COL_COLUMN_NAME);
+        builder.addPartitionKey(columnName);
+      }
+
+      if (!primaryKeyExists) {
+        throw new IllegalStateException("The table must have a primary key");
+      }
+
+      resultSet = metadata.getColumns(null, namespace, table, "%");
```

Comment:
Just a quick question. Clustering keys aren't automatically created in this import feature, right?

Reply:
No, they aren't. Because the current table-importing feature targets relational databases only, clustering keys no longer be needed under the relational abstraction.